// bootstrap-vue-next/dist/index-CdthQj6J.mjs
import { _ as _sfc_main, a as _sfc_main$1 } from "./BAccordion.vue_vue_type_script_setup_true_lang-BnfizlLE.mjs";
import { _ as _sfc_main$2 } from "./BAlert.vue_vue_type_script_setup_true_lang-Ddy_SctJ.mjs";
import { _ as _sfc_main$3, a as _sfc_main$4 } from "./BAvatarGroup.vue_vue_type_script_setup_true_lang-JninP8LV.mjs";
import { _ as _sfc_main$5 } from "./BBadge.vue_vue_type_script_setup_true_lang-BozDyuht.mjs";
import { _ as _sfc_main$6, a as _sfc_main$7 } from "./BBreadcrumb.vue_vue_type_script_setup_true_lang-C2qXYQvo.mjs";
import { _ as _sfc_main$8 } from "./BButton.vue_vue_type_script_setup_true_lang-DOxlfbAf.mjs";
import { _ as _sfc_main$9, a as _sfc_main$a } from "./BButtonToolbar.vue_vue_type_script_setup_true_lang-BSwHiGXg.mjs";
import { _ as _sfc_main$m } from "./BCloseButton.vue_vue_type_script_setup_true_lang-DCcXsu_x.mjs";
import { _ as _sfc_main$b, a as _sfc_main$c, b as _sfc_main$d, c as _sfc_main$f, d as _sfc_main$g, e as _sfc_main$h, f as _sfc_main$j } from "./BCard.vue_vue_type_script_setup_true_lang-BiNx3F25.mjs";
import { _ as _sfc_main$e, a as _sfc_main$i } from "./BCardText.vue_vue_type_script_setup_true_lang-DdqzUfNr.mjs";
import { _ as _sfc_main$k, a as _sfc_main$l } from "./BCarouselSlide.vue_vue_type_script_setup_true_lang-DWlh2imD.mjs";
import { _ as _sfc_main$o } from "./BCollapse.vue_vue_type_script_setup_true_lang-BMtSwwze.mjs";
import { _ as _sfc_main$n } from "./BCol.vue_vue_type_script_setup_true_lang-DrMYlpyB.mjs";
import { _ as _sfc_main$p } from "./BContainer.vue_vue_type_script_setup_true_lang-D1yL7duP.mjs";
import { _ as _sfc_main$1k } from "./BRow.vue_vue_type_script_setup_true_lang-B2V7AEcP.mjs";
import { _ as _sfc_main$q } from "./BDropdown.vue_vue_type_script_setup_true_lang-9RRC55vy.mjs";
import { _ as _sfc_main$r, a as _sfc_main$s, b as _sfc_main$t, c as _sfc_main$u, d as _sfc_main$v, e as _sfc_main$w, f as _sfc_main$x } from "./BDropdownText.vue_vue_type_script_setup_true_lang-leTd9fLR.mjs";
import { _ as _sfc_main$y } from "./BForm.vue_vue_type_script_setup_true_lang-B9irEupU.mjs";
import { _ as _sfc_main$B, a as _sfc_main$D } from "./BFormFloatingLabel.vue_vue_type_script_setup_true_lang-B8Wb_FR9.mjs";
import { _ as _sfc_main$G, a as _sfc_main$J, b as _sfc_main$Q, c as _sfc_main$S } from "./BFormValidFeedback.vue_vue_type_script_setup_true_lang-Beu8RDb6.mjs";
import { _ as _sfc_main$z, a as _sfc_main$A } from "./BFormCheckboxGroup.vue_vue_type_script_setup_true_lang-BjRIeRdy.mjs";
import { _ as _sfc_main$C } from "./BFormFile.vue_vue_type_script_setup_true_lang-BXGUOJRS.mjs";
import { _ as _sfc_main$E } from "./BFormGroup.vue_vue_type_script_setup_true_lang-ZoiXfQVL.mjs";
import { _ as _sfc_main$F } from "./BFormInput.vue_vue_type_script_setup_true_lang-DQpVGYwp.mjs";
import { _ as _sfc_main$H, a as _sfc_main$I } from "./BFormRadioGroup.vue_vue_type_script_setup_true_lang-JMMBIT4Y.mjs";
import { _ as _sfc_main$K, a as _sfc_main$M } from "./BFormSelect.vue_vue_type_script_setup_true_lang-ZOP30R97.mjs";
import { _ as _sfc_main$L } from "./useFormSelect-D2FHF0O9.mjs";
import { _ as _sfc_main$N } from "./BFormSpinbutton.vue_vue_type_script_setup_true_lang-I2tSKXQJ.mjs";
import { a as _sfc_main$O, _ as _sfc_main$P } from "./BFormTags.vue_vue_type_script_setup_true_lang-ZGc9xR1L.mjs";
import { _ as _sfc_main$R } from "./BFormTextarea.vue_vue_type_script_setup_true_lang-C28zkPR6.mjs";
import { _ as _sfc_main$T } from "./BImg.vue_vue_type_script_setup_true_lang-DmWhxDjR.mjs";
import { _ as _sfc_main$U, a as _sfc_main$V } from "./BInputGroupText.vue_vue_type_script_setup_true_lang-CnIjPlIE.mjs";
import { _ as _sfc_main$W } from "./BLink.vue_vue_type_script_setup_true_lang-D3KVrzdw.mjs";
import { _ as _sfc_main$X, a as _sfc_main$Y } from "./BListGroupItem.vue_vue_type_script_setup_true_lang-CNNcHiCb.mjs";
import { _ as _sfc_main$Z, a as _sfc_main$_ } from "./BModalOrchestrator.vue_vue_type_script_setup_true_lang-DsnyGGMw.mjs";
import { _ as _sfc_main$$, a as _sfc_main$10, b as _sfc_main$11, c as _sfc_main$12, d as _sfc_main$13 } from "./BNavText.vue_vue_type_script_setup_true_lang-ibc4Io6x.mjs";
import { _ as _sfc_main$14, a as _sfc_main$15, b as _sfc_main$16, c as _sfc_main$17 } from "./BNavbarToggle.vue_vue_type_script_setup_true_lang-BGOZ0d7l.mjs";
import { _ as _sfc_main$18 } from "./BOffcanvas.vue_vue_type_script_setup_true_lang-Bar1DMWR.mjs";
import { _ as _sfc_main$19 } from "./BOverlay.vue_vue_type_script_setup_true_lang-BHTZc193.mjs";
import { _ as _sfc_main$1a } from "./BPagination.vue_vue_type_script_setup_true_lang-CmTWzbW5.mjs";
import { _ as _sfc_main$1b, a as _sfc_main$1c, b as _sfc_main$1d, c as _sfc_main$1e, d as _sfc_main$1f } from "./BPlaceholderWrapper.vue_vue_type_script_setup_true_lang-DScaikV-.mjs";
import { _ as _sfc_main$1g } from "./floatingUi-BkI6EwT8.mjs";
import { _ as _sfc_main$1h } from "./BPopoverOrchestrator.vue_vue_type_script_setup_true_lang-DEdNfciC.mjs";
import { _ as _sfc_main$1i, a as _sfc_main$1j } from "./BProgress.vue_vue_type_script_setup_true_lang-PEKaHqrE.mjs";
import { _ as _sfc_main$1l } from "./BSpinner.vue_vue_type_script_setup_true_lang-D_xzH5Zg.mjs";
import { _ as _sfc_main$1n, a as _sfc_main$1o, b as _sfc_main$1r, c as _sfc_main$1s, d as _sfc_main$1t, e as _sfc_main$1u, f as _sfc_main$1v, g as _sfc_main$1z } from "./BTable.vue_vue_type_script_setup_true_lang-DhSnl763.mjs";
import { _ as _sfc_main$1p } from "./BTableSimple.vue_vue_type_script_setup_true_lang-D2TvxSs4.mjs";
import { _ as _sfc_main$1m, a as _sfc_main$1q } from "./BTabs.vue_vue_type_script_setup_true_lang-DU7z_v71.mjs";
import { _ as _sfc_main$1w, a as _sfc_main$1x } from "./BToastOrchestrator.vue_vue_type_script_setup_true_lang-DMlogLMR.mjs";
import { _ as _sfc_main$1y } from "./BTooltip.vue_vue_type_script_setup_true_lang-DYX8l-Iw.mjs";
const index = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  BAccordion: _sfc_main,
  BAccordionItem: _sfc_main$1,
  BAlert: _sfc_main$2,
  BAvatar: _sfc_main$3,
  BAvatarGroup: _sfc_main$4,
  BBadge: _sfc_main$5,
  BBreadcrumb: _sfc_main$6,
  BBreadcrumbItem: _sfc_main$7,
  BButton: _sfc_main$8,
  BButtonGroup: _sfc_main$9,
  BButtonToolbar: _sfc_main$a,
  BCard: _sfc_main$b,
  BCardBody: _sfc_main$c,
  BCardFooter: _sfc_main$d,
  BCardGroup: _sfc_main$e,
  BCardHeader: _sfc_main$f,
  BCardImg: _sfc_main$g,
  BCardSubtitle: _sfc_main$h,
  BCardText: _sfc_main$i,
  BCardTitle: _sfc_main$j,
  BCarousel: _sfc_main$k,
  BCarouselSlide: _sfc_main$l,
  BCloseButton: _sfc_main$m,
  BCol: _sfc_main$n,
  BCollapse: _sfc_main$o,
  BContainer: _sfc_main$p,
  BDropdown: _sfc_main$q,
  BDropdownDivider: _sfc_main$r,
  BDropdownForm: _sfc_main$s,
  BDropdownGroup: _sfc_main$t,
  BDropdownHeader: _sfc_main$u,
  BDropdownItem: _sfc_main$v,
  BDropdownItemButton: _sfc_main$w,
  BDropdownText: _sfc_main$x,
  BForm: _sfc_main$y,
  BFormCheckbox: _sfc_main$z,
  BFormCheckboxGroup: _sfc_main$A,
  BFormDatalist: _sfc_main$B,
  BFormFile: _sfc_main$C,
  BFormFloatingLabel: _sfc_main$D,
  BFormGroup: _sfc_main$E,
  BFormInput: _sfc_main$F,
  BFormInvalidFeedback: _sfc_main$G,
  BFormRadio: _sfc_main$H,
  BFormRadioGroup: _sfc_main$I,
  BFormRow: _sfc_main$J,
  BFormSelect: _sfc_main$K,
  BFormSelectOption: _sfc_main$L,
  BFormSelectOptionGroup: _sfc_main$M,
  BFormSpinbutton: _sfc_main$N,
  BFormTag: _sfc_main$O,
  BFormTags: _sfc_main$P,
  BFormText: _sfc_main$Q,
  BFormTextarea: _sfc_main$R,
  BFormValidFeedback: _sfc_main$S,
  BImg: _sfc_main$T,
  BInput: _sfc_main$F,
  BInputGroup: _sfc_main$U,
  BInputGroupText: _sfc_main$V,
  BLink: _sfc_main$W,
  BListGroup: _sfc_main$X,
  BListGroupItem: _sfc_main$Y,
  BModal: _sfc_main$Z,
  BModalOrchestrator: _sfc_main$_,
  BNav: _sfc_main$$,
  BNavForm: _sfc_main$10,
  BNavItem: _sfc_main$11,
  BNavItemDropdown: _sfc_main$12,
  BNavText: _sfc_main$13,
  BNavbar: _sfc_main$14,
  BNavbarBrand: _sfc_main$15,
  BNavbarNav: _sfc_main$16,
  BNavbarToggle: _sfc_main$17,
  BOffcanvas: _sfc_main$18,
  BOverlay: _sfc_main$19,
  BPagination: _sfc_main$1a,
  BPlaceholder: _sfc_main$1b,
  BPlaceholderButton: _sfc_main$1c,
  BPlaceholderCard: _sfc_main$1d,
  BPlaceholderTable: _sfc_main$1e,
  BPlaceholderWrapper: _sfc_main$1f,
  BPopover: _sfc_main$1g,
  BPopoverOrchestrator: _sfc_main$1h,
  BProgress: _sfc_main$1i,
  BProgressBar: _sfc_main$1j,
  BRow: _sfc_main$1k,
  BSpinner: _sfc_main$1l,
  BTab: _sfc_main$1m,
  BTable: _sfc_main$1n,
  BTableLite: _sfc_main$1o,
  BTableSimple: _sfc_main$1p,
  BTabs: _sfc_main$1q,
  BTbody: _sfc_main$1r,
  BTd: _sfc_main$1s,
  BTfoot: _sfc_main$1t,
  BTh: _sfc_main$1u,
  BThead: _sfc_main$1v,
  BToast: _sfc_main$1w,
  BToastOrchestrator: _sfc_main$1x,
  BTooltip: _sfc_main$1y,
  BTr: _sfc_main$1z
}, Symbol.toStringTag, { value: "Module" }));
export {
  index as i
};
